cli/filelister.cpp

```diff
@@ -193,34 +196,41 @@ static std::string addFiles2(std::map<std::string, std::size_t> &files,
 #endif
                 if (path_is_directory) {
                     if (recursive && !ignored.match(new_path)) {
-                        std::string err = addFiles2(files, new_path, extra, recursive, ignored);
+                        std::string err = addFiles2(filesSorted, new_path, extra, recursive, ignored);
                         if (!err.empty()) {
                             return err;
                         }
                     }
                 } else {
                     if (Path::acceptFile(new_path, extra) && !ignored.match(new_path)) {
-                        if (stat(new_path.c_str(), &file_stat) != -1)
-                            files[new_path] = file_stat.st_size;
+                        if (stat(new_path.c_str(), &file_stat) != -1) {
+                            filesSorted.emplace_back(new_path, file_stat.st_size);
+                        }
                         else {
                             const int err = errno;
                             return "could not stat file '" + new_path + "' (errno: " + std::to_string(err) + ")";
                         }
                     }
                 }
             }
+
+            filesSorted.sort([](const decltype(filesSorted)::value_type& a, const decltype(filesSorted)::value_type& b) {
```

Comment:
The files inside the folder still should be sorted as the order retrieved from the filesystem is random.

I did not add an explicit test for this as this is implicitly tested by `dmake`. The order of the files were different in the generated files before I added the sorting.